deployment/data-streams/changeset/sequence/deploy_data_streams_test.go

```diff
@@ -0,0 +1,151 @@
+)
+
+var (
+	verificationCfg = verification.SetConfig{
+		ConfigDigest: [32]byte{1},
+		Signers: []common.Address{
+			common.HexToAddress("0x1111111111111111111111111111111111111111"),
+			common.HexToAddress("0x2222222222222222222222222222222222222222"),
+			common.HexToAddress("0x3333333333333333333333333333333333333333"),
+			common.HexToAddress("0x4444444444444444444444444444444444444444"),
+		},
+		F: 1,
+	}
+)
+
+func TestDeployDataStreamsContracts(t *testing.T) {
+	tests := []struct {
+		name                    string
+		hasExistingMcms         bool
+		deployDataStreamsConfig DeployDataStreamsConfig
+		expectedContracts       []deployment.ContractType
+	}{
+		{
+			name:            "Deploy with billing and MCMS",
+			hasExistingMcms: false,
+			deployDataStreamsConfig: DeployDataStreamsConfig{
+				ChainsToDeploy: map[uint64]DeployDataStreams{testutil.TestChain.Selector: {
+					VerifierConfig: verificationCfg,
+					Billing: BillingFeature{
+						Enabled: true,
+						Config: &BillingConfig{
+							NativeTokenAddress: common.HexToAddress("0x3e5e9111ae8eb78fe1cc3bb8915d5d461f3ef9a9"),
+						},
+					},
+					Ownership: types.OwnershipFeature{
+						Transfer:           true,
+						MCMSProposalConfig: &proposalutils.TimelockConfig{MinDelay: 0},
+						DeployMCMS:         true,
+						DeployMCMSConfig:   proposalutils.SingleGroupTimelockConfigV2(t),
+					},
+				}},
+			},
+			expectedContracts: []deployment.ContractType{types.VerifierProxy, types.Verifier, types.RewardManager, types.FeeManager,
+				commontypes.ProposerManyChainMultisig, commontypes.BypasserManyChainMultisig, commontypes.CancellerManyChainMultisig},
+		},
+		{
+			name:            "Deploy no billing and MCMS",
+			hasExistingMcms: false,
+			deployDataStreamsConfig: DeployDataStreamsConfig{
+				ChainsToDeploy: map[uint64]DeployDataStreams{testutil.TestChain.Selector: {
+					VerifierConfig: verificationCfg,
+					Ownership: types.OwnershipFeature{
+						Transfer:           true,
+						MCMSProposalConfig: &proposalutils.TimelockConfig{MinDelay: 0},
+						DeployMCMS:         true,
+						DeployMCMSConfig:   proposalutils.SingleGroupTimelockConfigV2(t),
+					},
+				}},
+			},
+			expectedContracts: []deployment.ContractType{types.VerifierProxy, types.Verifier,
+				commontypes.ProposerManyChainMultisig, commontypes.BypasserManyChainMultisig, commontypes.CancellerManyChainMultisig},
+		},
+		{
+			name:            "Deploy no billing with existing MCMS",
+			hasExistingMcms: true,
+			deployDataStreamsConfig: DeployDataStreamsConfig{
+				ChainsToDeploy: map[uint64]DeployDataStreams{testutil.TestChain.Selector: {
+					VerifierConfig: verificationCfg,
+					Ownership: types.OwnershipFeature{
+						Transfer:           true,
+						MCMSProposalConfig: &proposalutils.TimelockConfig{MinDelay: 0},
+					},
+				}},
+			},
+			expectedContracts: []deployment.ContractType{types.VerifierProxy, types.Verifier},
+		},
+		{
```

Comment:
We can add a test for when the sequence fail in the middle.

Reply:
This sequence itself doesn't have any durability guarantees - so there isn't much to test in a failure scenario